### ISSUE TITLE
[Bug] iOS - scroll/swipe issue with a SwipeView inside a ListView

### ISSUE THREAD

**Issue by ragu89** [xamarin/Xamarin.Forms#14462]
### Description

on iOS, the [SwipeView](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/swipeview) component doesn't work good when it's integrated in a [ListView](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/listview/). Specially when the [Caching Strategy](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/listview/performance#caching-strategy) is set to **RecycleElement**.

### Steps to Reproduce

1. Create a [ListView with Caching Strategy set to RecycleElement](https://github.com/ragu89/xam-list-of-swipeviews/blob/main/ListOfSwipeViews/MainPage.xaml#L8)
2. Insert SwipeView components inside the ListView
3. Run on iOS
4. Scroll right/left/down/up on the items of the List (monkey test)

It's also possible to reproduce when there is *no caching strategy*, but it's more difficult.

### Expected Behavior

The Swipe is handled correctly on the items.

### Actual Behavior

The items are move off the screen during the swipe with an unexpected offset.

### Basic Information

- Version with issue: 5.0.0.2012
- Platform Target Frameworks: 
  - iOS:  14.5
- Affected Devices: All iOS phone. Not tested on iPad.

### Screenshots

![issue-demo](https://user-images.githubusercontent.com/1267845/127122640-405ed2d1-4f5b-435a-b3db-c911a9604b8c.gif)

### Reproduction Link

- Demo project : https://github.com/ragu89/xam-list-of-swipeviews

**Comment by jis248** [comment]
Yes, I am also facing this issue. Is there any alternative solution for this.
